packages/labs/src/components/timezone-select/timezoneSelect.tsx

```diff
@@ -0,0 +1,509 @@
+        const isPlaceholder = !value;
+        const classes = classNames(Classes.TIMEZONE_SELECT_TARGET, targetClassName, {
+            [Classes.TIMEZONE_SELECT_TARGET_PLACEHOLDER]: isPlaceholder,
+        });
+
+        return (
+            <Button
+                className={classes}
+                text={displayValue || defaultDisplayValue || placeholder}
+                rightIconName="caret-down"
+                disabled={disabled}
+            />
+        );
+    }
+
+    private updateTimezones(date: Date): void {
+        const timezones = getTimezoneItems(date);
+        this.timezones = timezones;
+        this.timezoneToQueryCandidates = getTimezoneQueryCandidates(timezones, date);
+    }
+
+    private filterTimezones = (query: string, items: ITimezoneItem[]): ITimezoneItem[] => {
+        const normalizedQuery = normalizeText(query);
+        return items.filter((item) => {
+            const candidates = this.timezoneToQueryCandidates[item.timezone] || [normalizeText(item.timezone)];
+            return candidates.some((candidate) => isQueryMatch(normalizedQuery, candidate));
+        });
+    }
+
+    private renderItem = (itemProps: ISelectItemRendererProps<ITimezoneItem>) => {
+        const { item, isActive, handleClick } = itemProps;
+        const classes = classNames(CoreClasses.MENU_ITEM, CoreClasses.intentClass(), {
+            [CoreClasses.ACTIVE]: isActive,
+            [CoreClasses.INTENT_PRIMARY]: isActive,
+        });
+
+        return (
+            <MenuItem
+                key={item.key}
+                className={classes}
+                iconName={item.iconName}
+                text={item.text}
+                label={item.label}
+                onClick={handleClick}
+                shouldDismissPopover={false}
+            />
+        );
+    }
+
+    private handleItemSelect = (timezone: ITimezoneItem) => {
+        if (this.props.value === undefined) {
+            this.setState({ value: timezone.timezone });
+        }
+        Utils.safeInvoke(this.props.onChange, timezone.timezone);
+    }
+
+    private handleQueryChange = (query: string) => {
+        this.setState({ query });
+    }
+}
+
+function getTimezoneItems(date: Date): ITimezoneItem[] {
+    return moment.tz.names().map((timezone) => toTimezoneItem(timezone, date));
+}
+
+function getInitialTimezoneItems(date: Date, showLocalTimezone: boolean): ITimezoneItem[] {
+    const populous = getPopulousTimezoneItems(date);
+    const local = showLocalTimezone ? getLocalTimezoneItem(date) : undefined;
+    return local ? [local, ...populous] : populous;
+}
+
+function getPopulousTimezoneItems(date: Date): ITimezoneItem[] {
+    const timezones = moment.tz.names()
+        .filter((timezone) => (
+            // Filter out noisy timezones
+            // See https://github.com/moment/moment-timezone/issues/227
+            /\//.test(timezone) &&
+            !/Etc\//.test(timezone)
+        ));
```

Comment:
nit: two lines instead of 6
```js
// Filter out noisy timezones. See https://github.com/moment/moment-timezone/issues/227
.filter((timezone) => /\//.test(timezone) && !/Etc\//.test(timezone))
```